### PR TITLE
Remove entityType from CheckRealizeState

### PR DESCRIPTION
**Pull request by TaoZou1**
Remove entityType from CheckRealizeState. Since we're not sure if the realized 
items for intentPath will be changed or not

### CODE REVIEW

File: pkg/nsx/services/realizestate/realize_state.go [hunk 1]
```diff
@@ -46,10 +46,10 @@ func IsRealizeStateError(err error) bool {
 	return ok
 }
 
-// CheckRealizeState allows the caller to check realize status of entityType with retries.
-// backoff defines the maximum retries and the wait interval between two retries.
-// if entityType == "", check all the entities, all entities should be in the REALIZED state to be tereated as REALIZED
-func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, intentPath, entityType string) error {
+// CheckRealizeState allows the caller to check realize status of intentPath with retries.
+// Backoff defines the maximum retries and the wait interval between two retries.
+// Check all the entities, all entities should be in the REALIZED state to be tereated as REALIZED
```
Comment (yanjunz97) [inline pkg/nsx/services/realizestate/realize_state.go:51]:
```suggestion
// Check all the entities, all entities should be in the REALIZED state to be treated as REALIZED
```

**Review by yanjunz97 — COMMENTED** [review]
Overall LGTM, just found one typo

File: pkg/nsx/services/realizestate/realize_state.go [hunk 2]
```diff
@@ -62,9 +62,6 @@ func (service *RealizeStateService) CheckRealizeState(backoff wait.Backoff, inte
 		}
 		entitiesRealized := 0
 		for _, result := range results.Results {
-			if entityType != "" && result.EntityType != nil && *result.EntityType != entityType {
-				continue
-			}
 			if *result.State == model.GenericPolicyRealizedResource_STATE_REALIZED {
```
Comment (wenyingd) [inline pkg/nsx/services/realizestate/realize_state.go:65]: I prefer to do it like this with function. No need to explicitly compare the count of model.GenericPolicyRealizedResource_STATE_REALIZED resources. Instead, I assume the intent of this function is all NSX returned resources should be with `model.GenericPolicyRealizedResource_STATE_REALIZED` , then we can directly return error if any is not in this state. 

```
func() error {
		results, err := service.NSXClient.RealizedEntitiesClient.List(intentPath, nil)
		err = nsxutil.TransNSXApiError(err)
		if err != nil {
			return err
		}
		for _, result := range results.Results {
			if *result.State == model.GenericPolicyRealizedResource_STATE_ERROR {
				var errMsg []string
				for _, alarm := range result.Alarms {
					if alarm.Message != nil {
						errMsg = append(errMsg, *alarm.Message)
					}
				}
				return NewRealizeStateError(fmt.Sprintf("%s realized with errors: %s", *result.EntityType, errMsg))
			}
			if *result.State != model.GenericPolicyRealizedResource_STATE_REALIZED {
				return fmt.Errorf("%s not realized", entityType)
			}
		}
		return nil
	}
```

Comment (TaoZou1) [reply pkg/nsx/services/realizestate/realize_state.go:65]: It seems there is a problem.
If you have 3 items. One is unrealized, one is error, one is realized.
Your logic depends on the return order.

File: pkg/nsx/services/realizestate/realize_state.go (hunk 1 above)
Comment (TaoZou1) [reply pkg/nsx/services/realizestate/realize_state.go:51]: done

**Review by wenyingd — APPROVED** [review]

**Review by yanjunz97 — APPROVED** [review]

**Pull request merged**